batch/sql/cleanup-add-batch-updates.sql

```diff
@@ -0,0 +1,29 @@
+START TRANSACTION;
+
+ALTER TABLE jobs MODIFY COLUMN `update_id` INT NOT NULL;
+ALTER TABLE batches_inst_coll_staging MODIFY COLUMN `update_id` INT NOT NULL;
+ALTER TABLE batch_inst_coll_cancellable_resources MODIFY COLUMN `update_id` INT NOT NULL;
+
+ALTER TABLE jobs ADD FOREIGN KEY (`batch_id`, `update_id`) REFERENCES batch_updates (`batch_id`, `update_id`) ON DELETE CASCADE;
+ALTER TABLE batches_inst_coll_staging ADD FOREIGN KEY (`batch_id`, `update_id`) REFERENCES batch_updates (`batch_id`, `update_id`) ON DELETE CASCADE;
+ALTER TABLE batch_inst_coll_cancellable_resources ADD FOREIGN KEY (`batch_id`, `update_id`) REFERENCES batch_updates (`batch_id`, `update_id`) ON DELETE CASCADE;
+
+SET foreign_key_checks = 0;
```

Comment:
what's this about?

Reply:
I was under the impression that dropping the primary key (even if it is to be immediately replaced) could error while foreign key checks were enabled since it would temporarily invalidate other tables' foreign keys. @jigold Is that correct?

Reply:
I can't remember why. It could be copy / paste from a slightly different scenario with the resource id change. I'd say remove it and see if you get an error and then the error might illuminate what the issue was.

Reply:
Found the issue.
https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl-operations.html#online-ddl-foreign-key-operations

Adding a foreign key constraint without key checks set to 0 means the operation isn't able to COPY INPLACE. I'll check tomorrow whether the table is locked while this operation is going on. My hope is it doesn't.